jbmc/src/java_bytecode/java_utils.cpp

```diff
@@ -510,3 +511,41 @@ void set_declaring_class(symbolt &symbol, const irep_idt &declaring_class)
 {
   symbol.type.set(ID_C_class, declaring_class);
 }
+
+static const symbolt *find_cprover_nondet_initialize_on_class(
+  const symbol_tablet &symbol_table,
+  const std::string &class_name)
+{
+  const irep_idt method_name(class_name + ".cproverNondetInitialize:()V");
+  const auto ret = symbol_table.symbols.find(method_name);
+  if(ret != symbol_table.symbols.end())
+  {
+    return &ret->second;
+  }
+  return nullptr;
+}
+
+const symbolt *find_cprover_nondet_initialize(
```

Comment:
Is `optionalt` usable here? I know it causes problems in certain situations (sadly), so wouldn't be surprised if the answer is no.

Reply:
You can't have an `optionalt` of a reference. They say to use a pointer. I could use optionalt of a reference wrapper, I think.